src/app/clusters/refrigerator-alarm-server/refrigerator-alarm-server.cpp

```diff
@@ -0,0 +1,188 @@
+
+    return status;
+}
+
+EmberAfStatus RefrigeratorAlarmServer::getStateValue(chip::EndpointId endpoint, chip::BitMask<AlarmMap> * state)
+{
+    EmberAfStatus status = Attributes::State::Get(endpoint, state);
+    if (status != EMBER_ZCL_STATUS_SUCCESS)
+    {
+        emberAfPrintln(EMBER_AF_PRINT_DEBUG, "Refrigerator Alarm: ERR: reading state %x", status);
+    }
+
+    emberAfPrintln(EMBER_AF_PRINT_DEBUG, "Refrigerator Alarm: State ep%d value: %" PRIu32 "", endpoint, state->Raw());
+
+    return status;
+}
+
+EmberAfStatus RefrigeratorAlarmServer::setMaskValue(chip::EndpointId endpoint, chip::BitMask<AlarmMap> mask)
+{
+    chip::BitMask<AlarmMap> * currentMask = nullptr;
+    EmberAfStatus status                  = Attributes::Mask::Get(endpoint, currentMask);
+    if (status != EMBER_ZCL_STATUS_SUCCESS)
+    {
+        emberAfPrintln(EMBER_AF_PRINT_DEBUG, "Refrigerator Alarm: ERR: reading  mask %x", status);
+    }
+    if (currentMask->Raw() == mask.Raw())
+    {
+        emberAfPrintln(EMBER_AF_PRINT_DEBUG, "Refrigerator Alarm: Mask value already set to %" PRIu32 "", mask.Raw());
+    }
+    else
+    {
+        status = Attributes::Mask::Set(endpoint, mask);
+        if (status != EMBER_ZCL_STATUS_SUCCESS)
+        {
+            emberAfPrintln(EMBER_AF_PRINT_DEBUG, "Refrigerator Alarm: ERR: writing  mask %x", status);
+        }
+    }
+
+    emberAfPrintln(EMBER_AF_PRINT_DEBUG, "Refrigerator Alarm: Mask ep%d value: %" PRIu32 "", endpoint, mask.Raw());
+
+    return status;
+}
+
+bool RefrigeratorAlarmServer::SendNotifyEvent(chip::EndpointId endpointId, chip::BitMask<AlarmMap> active,
+                                              chip::BitMask<AlarmMap> inActive, chip::BitMask<AlarmMap> state,
+                                              chip::BitMask<AlarmMap> mask)
+{
+    Events::Notify::Type event{ active, inActive, state, mask };
+    EventNumber eventNumber;
+    CHIP_ERROR error = app::LogEvent(event, endpointId, eventNumber);
+    if (CHIP_NO_ERROR != error)
+    {
+        ChipLogError(Zcl, "[Notify] Unable to send notify event: %s [endpointId=%d]", error.AsString(), endpointId);
+        return false;
+    }
+
+    return true;
+}
+
+bool RefrigeratorAlarmServer::ResetCommand(CommandHandler * commandObj, const ConcreteCommandPath & commandPath,
+                                           const Commands::Reset::DecodableType & commandData)
+{
+    auto & alarms = commandData.alarms;
+    auto & mask   = commandData.mask;
+
+    EndpointId endpoint = commandPath.mEndpointId;
+
+    EmberAfStatus status;
+
+    chip::BitMask<AlarmMap> eActive   = 0;
+    chip::BitMask<AlarmMap> eInActive = alarms;
+    chip::BitMask<AlarmMap> eMask     = 0;
+    chip::BitMask<AlarmMap> eState    = 0;
+
+    chip::BitMask<AlarmMap> state = 0;
+    status                        = State::Get(endpoint, &state);
+
+    eActive.Clear(alarms);
```

Comment:
eActive already has everything cleared.  This logic does not make sense.  In fact, nothing ever becomes active _in this function_.

Reply:
Updated logic acc to spec